filecoin-proofs/tests/api.rs

```diff
@@ -1833,6 +1848,23 @@ fn compare_elements(path1: &Path, path2: &Path) -> Result<(), Error> {
     Ok(())
 }
 
+/// Compare the elements of trees in two different directories. If the tree is split into multiple
+/// files, only the first file is checked.
+fn compare_elements_with_cache_key<Tree: 'static + MerkleTreeTrait>(
```

Comment:
The purpose of this function is just to check equivalency of the first base trees in `dir_a` and `dib_b`? Is that an intentional constraint (maybe for performance reasons) or does it make sense for the function to check full tree equivalency?

Reply:
For performance reasons I thought it's good enough to check the first file. If the result is different, I'd expect it to be wildly different, hence already on the first file.

Reply:
I'm agreeing with Jake here, for consistency, it makes sense to check the entire tree.  Test sector sizes are very quick to check, so it can be avoided at prod sector sizes in tests where possible (although those should be on the order of a few minutes in any case)

Reply:
Would it make sense to introduce file hashing instead? Given that's really about "the files are equal", that they are correct field elements was already checked.

Reply:
I've added hashing the files.

Reply:
Yup, works for me.